internal/pkg/service/common/task/result.go

```diff
@@ -3,13 +3,25 @@ package task
 import (
 	"strings"
 
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 )
 
 type Result struct {
-	result  string
-	error   error
-	outputs map[string]any
+	result          string
+	error           error
+	errorType       string
+	unexpectedError bool
```

Comment:
V background taskoch potrebujem odlisit `expectedError` - equvalent `4xx` pri HTTP, a `unexpectedError` = `5xx` pri HTTP.

`expectedError` bude napr. pri BA, ze uzivatel si sam refreshol token a s tym nic nevieme spravit, ale nema sa to pocitat medzi chyby a spravit incident.

Reply:
V PHP to rozlišujeme na User a Application exception https://github.com/keboola/common-exceptions/tree/master/src úplně nevím jestli když to uvidím v telemetrii tak jestli vím, co to znamená expectedError, ale podle toho co píšeš tak je to ekvivalentní našeho UserException tak bych sa možná držal toho co používáme napříč platformů.